src/BinSkim.Driver/AnalyzeCommand.cs

```diff
@@ -43,6 +43,8 @@ protected override BinaryAnalyzerContext CreateContext(AnalyzeOptions options, I
                 ShouldWarnVerbose = false;
             }
 
+            binaryAnalyzerContext.CompilerDataLogger = new CompilerDataLogger(binaryAnalyzerContext, options.RepositoryUri, options.PipelineName);
```

Comment:
>CompilerDataLogger [](http://example.com/codeflow?start=34&length=18)

Why do we always create this? Instead of conditionally?

Reply:
I'm creating every time  because only here i will have all  the information (pipeline, repository).

the appinsights part is static, so it wont keep re-creating it.

Reply:
added some conditions to prevent creating this class and not using.
thanks for the feedback.